app/models/user.rb

```diff
@@ -53,6 +64,51 @@ def send_activation_email
     UserMailer.account_activation(self).deliver_now
   end
 
+  # パスワード再設定の属性を設定する
+  def create_reset_digest
+    self.reset_token = User.new_token
+    update_columns(reset_digest: User.digest(reset_token), reset_sent_at: Time.zone.now)
+  end
+
+  # パスワード再設定のメールを送信する
+  def send_password_reset_email
+    UserMailer.password_reset(self).deliver_now
+  end
+
+  # パスワード再設定の期限が切れている場合はtrueを返す
+  def password_reset_expired?
+    reset_sent_at < 2.hours.ago
+  end
+
+  # ユーザーのステータスフィードを返す
+  def feed
+    following_ids = "SELECT followed_id FROM relationships
+                     WHERE follower_id = :user_id"
+    Micropost.where("user_id IN (#{following_ids})
+                     OR user_id = :user_id", user_id: id)
```

Comment:
```suggestion
    Micropost.where(user_id: following_ids).or(Micropost.where(user_id: id))
```
ドキュメント欄にも記載してくれていますが、よりシンプルにこういった書き方もできます。
5系から追加されました。
https://style.potepan.com/articles/29804.html